dan_layer/wallet/storage_sqlite/src/reader.rs

```diff
@@ -665,29 +668,68 @@ impl WalletStoreReader for ReadTransaction<'_> {
             .optional()
             .map_err(|e| WalletStorageError::general("accounts_get_by_vault", e))?
             .ok_or_else(|| WalletStorageError::NotFound {
-                operation: "get_non_fungible_token",
+                operation: "non_fungible_token_get_by_nft_id",
                 entity: "non_fungible_tokens".to_string(),
                 key: format!("{}", vault_id),
             })?;
         let vault_address = VaultId::from_str(&vault_address).map_err(|e| WalletStorageError::DecodingError {
             details: e.to_string(),
             item: "non_fungible_tokens",
-            operation: "get_non_fungible_token",
+            operation: "non_fungible_token_get_by_nft_id",
         })?;
         non_fungible_token.try_into_non_fungible_token(vault_address)
     }
 
-    fn get_resource_address(&mut self, nft_id: NonFungibleId) -> Result<ResourceAddress, WalletStorageError> {
+    fn non_fungible_token_get_all(
+        &mut self,
+        limit: u64,
+        offset: u64,
+    ) -> Result<Vec<NonFungibleToken>, WalletStorageError> {
+        use crate::schema::{non_fungible_tokens, vaults};
+
+        let non_fungibles = non_fungible_tokens::table
+            .limit(limit as i64)
+            .offset(offset as i64)
+            .load::<models::NonFungibleToken>(self.connection())
+            .map_err(|e| WalletStorageError::general("non_fungible_token_get_all", e))?;
+
+        let vault_ids = non_fungibles.iter().map(|n| n.vault_id).collect::<Vec<_>>();
+        let vault_addresses = vaults::table
+            .select(vaults::address)
+            .filter(vaults::id.eq_any(vault_ids))
+            .load::<String>(self.connection())
+            .map_err(|e| WalletStorageError::general("accounts_get_by_vault", e))?;
```

Comment:
neat trick: eq_any also takes an iterator so no need to collect it into a buffer first
```suggestion
        let vault_ids = non_fungibles.iter().map(|n| n.vault_id);
        let vault_addresses = vaults::table
            .select(vaults::address)
            .filter(vaults::id.eq_any(vault_ids))
            .load::<String>(self.connection())
            .map_err(|e| WalletStorageError::general("accounts_get_by_vault", e))?;
```

Reply:
Cool ! Thanks for the hint :)